qbft/json_testutils.go

```diff
@@ -52,38 +52,91 @@ func (i *Instance) GetRoot() ([32]byte, error) {
 	return i.State.GetRoot()
 }
 
+func (c *Controller) MarshalJSON() ([]byte, error) {
+
+	// Create alias without config due to custom marshalling error
+	type ControllerAlias struct {
+		Identifier      []byte
+		Height          Height
+		StoredInstances InstanceContainer
+		CommitteeMember *types.CommitteeMember
+		OperatorSigner  *types.OperatorSigner `json:"-"`
+	}
+
+	alias := &ControllerAlias{
+		Identifier:      c.Identifier,
+		Height:          c.Height,
+		StoredInstances: c.StoredInstances,
+		CommitteeMember: c.CommitteeMember,
+		OperatorSigner:  c.OperatorSigner,
+	}
+
+	return json.Marshal(alias)
+}
+
+func (c *Controller) UnmarshalJSON(data []byte) error {
+
+	// Create alias without config due to custom marshalling error
+	type ControllerAlias struct {
+		Identifier      []byte
+		Height          Height
+		StoredInstances InstanceContainer
+		CommitteeMember *types.CommitteeMember
+		OperatorSigner  *types.OperatorSigner `json:"-"`
+	}
+
+	aux := &ControllerAlias{}
+
+	// Unmarshal the JSON data into the auxiliary struct
+	if err := json.Unmarshal(data, &aux); err != nil {
+		return err
+	}
+
+	c.Identifier = aux.Identifier
+	c.Height = aux.Height
+	c.StoredInstances = aux.StoredInstances
+	c.CommitteeMember = aux.CommitteeMember
+	c.OperatorSigner = aux.OperatorSigner
+
+	return nil
+}
```

Comment:
Before, `config` was private (first letter lowercase), so it didn't appear in JSONs. With `GetConfig` being dropped and `config`$\to$`Config` switching to a public attribute, it started appearing in JSONs. However, it can't be unmarshalled due to its function types. So, it was necessary to hide the config in the custom marshalling.

Note: in tests, we fill the `Config` attribute with the default testing values.

Right @alan-ssvlabs ?

Reply:
Totally correct, thanks Matheus

Reply:
It is better to add `json:"-"` next to config to make serialization ignore this field and remove the extra code

Reply:
Didn't know this can work, updated, thanks!

Reply:
Didn't know this can work, updated, thanks!